### PR TITLE
Bug 1848885: check if 404 returned before ask users questions

### PR DESCRIPTION
**Pull request by dougsland**
Before ask users questions like: user/pass we should
validate that host AND PKI URL from engine are both
reachable.

Bug-Url: https://bugzilla.redhat.com/show_bug.cgi?id=1848885
Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

### CODE REVIEW

File: pkg/asset/installconfig/ovirt/credentials.go [hunk 1]
```diff
@@ -48,24 +48,31 @@ func (c *clientHTTP) downloadFile() error {
...
 	if c.saveFilePath == "" {
-		return errors.New("saveFilePath must be specificed")
+		return httpError404, errors.New("saveFilePath must be specificed")
 	}
 
 	client := &http.Client{Transport: tr}
 	resp, err := client.Get(c.urlAddr)
+
+	switch resp.StatusCode {
+	case httpError404:
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/ovirt/credentials.go:58]: https://golang.org/pkg/net/http/#pkg-constants `StatusNotFound`

File: pkg/asset/installconfig/ovirt/credentials.go [hunk 2]
```diff
@@ -48,24 +48,31 @@ func (c *clientHTTP) downloadFile() error {
...
+		msg := fmt.Sprintf("http response 404 for: %s", c.urlAddr)
+		return resp.StatusCode, errors.New(msg)
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/ovirt/credentials.go:60]: why the msg and return errors.New(msg), maybe just use `https://godoc.org/github.com/pkg/errors#Errorf`

Comment (dougsland) [reply pkg/asset/installconfig/ovirt/credentials.go:60]: thanks!

File: pkg/asset/installconfig/ovirt/credentials.go (hunk 1 above)
Comment (dougsland) [reply pkg/asset/installconfig/ovirt/credentials.go:58]: great, thanks.

File: pkg/asset/installconfig/ovirt/credentials.go [hunk 3]
```diff
@@ -48,24 +48,30 @@ func (c *clientHTTP) downloadFile() error {
 	}
 
 	if c.saveFilePath == "" {
-		return errors.New("saveFilePath must be specificed")
+		return http.StatusNotFound, errors.New("saveFilePath must be specificed")
```
[the remark this answers is not in the document]
Comment (dougsland) [reply pkg/asset/installconfig/ovirt/credentials.go:51]: Thanks for your suggestion @rgolangh.  We would need to create a var + Wrap for each error we would like to add/return. 
Right now, we are using return resp.StatusCode, errors.Errorf("http response 404 for: %s", c.urlAddr) which seems modular and simple enough. Please let me know if you would like me to convert to the approach you described or continue with what we have now.

[the remark this answers is not in the document]
Comment (rgolangh) [reply pkg/asset/installconfig/ovirt/credentials.go:51]: > Thanks for your suggestion @rgolangh. We would need to create a var + Wrap for each error we would like to add/return.

You only need to wrap one. The rest of the errors could be left untouched.

> Right now, we are using return resp.StatusCode, errors.Errorf("http response 404 for: %s", 
c.urlAddr) 

The http 404 is not the real error so this is misleading. When I read it I was looking for the http request and couldn't find it.

**Comment by dougsland** [comment]
I've implemented Roy request and github opened a new PR https://github.com/openshift/installer/pull/3874 
Closing this one.

**Pull request closed without merging**